app/src/main/java/de/tum/in/tumcampusapp/api/app/ConnectivityInterceptor.kt

```diff
@@ -0,0 +1,23 @@
+package de.tum.`in`.tumcampusapp.api.app
+
+import android.content.Context
+import de.tum.`in`.tumcampusapp.api.app.exception.NoNetworkConnectionException
+import de.tum.`in`.tumcampusapp.utils.NetUtils
+import okhttp3.Interceptor
+import okhttp3.Response
+import java.io.IOException
+
+class ConnectivityInterceptor(private val context: Context) : Interceptor {
+
+    @Throws(IOException::class)
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+
+        if (!NetUtils.isConnected(context)) {
+            throw NoNetworkConnectionException()
```

Comment:
do we actually handle that exception somewhere?

Reply:
Not by us. When `NoNetworkConnectionException` is thrown, it’s caught and passed to Retrofit’s `onFailure()` callback and can be handled there, whereas previously it would result in an uncaught exception. 

Every caller of Retrofit can handle network exceptions themselves. Right now, we mostly only log those.

Reply:
So why does this then fix the crashes, if we just throw other exceptions?
Are IOExceotions already handled by OkHttp?